core/base/inc/TString.h

```diff
@@ -166,6 +169,14 @@ operator+(T f, const TString &s);
 
 friend Bool_t  operator==(const TString &s1, const TString &s2);
 friend Bool_t  operator==(const TString &s1, const char *s2);
+#ifdef __cpp_lib_three_way_comparison
+friend std::strong_ordering operator<=>(const TString &s1, const TString &s2) {
```

Comment:
Why are we using `#ifdef __cpp_lib_three_way_comparison` here and `#if (__cplusplus >=202002L)` above.  I think `std::strong_ordering` should be defined anyways since C++20, no?
Otherwise, LGTM.

Reply:
We wanted to avoid `#if (__cplusplus >=202002L)`, but at the end it was needed for the `#include <compare>`, to prevent the C++ modules to pick it even when compiling without `-std=c++20`. I.e. the code below was making the c++ modules complaining about `<compare>` being included without `-std=c++20`.
```
#if __has_include(<compare>)
#  include <compare>
#endif
```
(you can just parse the history of this PR to see the several iterations...)
But I abree now we could use `#if (__cplusplus >=202002L)` everywhere, but I prefer the way it is now nevertheless

Reply:
> (you can just parse the history of this PR to see the several iterations...)
> But I agree now we could use `#if (__cplusplus >=202002L)` everywhere, but I prefer the way it is now nevertheless

I skimmed through the PR history in the past few days, but I was just curious as to why here we use a different check (and not the same above, i.e. `#if (__cplusplus >=202002L)`).

> But I agree now we could use `#if (__cplusplus >=202002L)` everywhere, but I prefer the way it is now nevertheless

Fair enough.  :slightly_smiling_face:

Reply:
> But I agree now we could use `#if (__cplusplus >=202002L)` everywhere, but I prefer the way it is now nevertheless

The only problem might be with a compiler that implements `__cpp_lib_three_way_comparison` already in C++17 - but let's take that risk :-)